cargo-miri/bin.rs

```diff
@@ -652,84 +698,93 @@ fn phase_cargo_runner(binary: &Path, binary_args: env::Args) {
     let file = File::open(&binary)
         .unwrap_or_else(|_| show_error(format!("file {:?} not found or `cargo-miri` invoked incorrectly; please only invoke this binary through `cargo miri`", binary)));
     let file = BufReader::new(file);
-    let info: CrateRunInfo = serde_json::from_reader(file)
-        .unwrap_or_else(|_| show_error(format!("file {:?} contains outdated or invalid JSON; try `cargo clean`", binary)));
```

Comment:
Please keep the separate `let`-binding; that's much more readable than the long function call in the `match` operand.